[MDS-5353] ESUP Modal updates

matbusby-fw opened the pull request:
- Built view esup modal
- updated action menu on esup table
- added tables in view and edit modals to show generated documents and support documents
- styled both modals

Lot's of styling stuff here.
Decided to have a seperate view modal since it can be pretty simplified and static, while the edit modal looks like it will have more conditions attached to it as we move forward.  A bit of repetition of things as a result, but I think it'll keep it more readable in the future.

Hit a pretty funky merge conflict that took some time to untangle on the explosives permit table.  Should be good though.

## Objective 

[MDS-](https://bcmines.atlassian.net/browse/MDS-0000)

![image](https://github.com/bcgov/mds/assets/83598933/78cd9ce3-e8c8-4684-bbca-f3855f233049)
![image](https://github.com/bcgov/mds/assets/83598933/306c665b-3624-46cb-a829-0534ee638757)
![image](https://github.com/bcgov/mds/assets/83598933/a97bbc81-a871-4357-8e59-0f2d468fe0df)

matbusby committed:
Add Explosives Permit form and Remove token after complete download

New form has been created to handle the addition of explosives permits. Form will handle business rules validations, state, and UI. Ant.design and redux-form have been used to handle state management and UI.

In addition, a minor bug fix was implemented in the document.py file to ensure that the cache for downloading documents get deleted after the download operation completes. This is to prevent reuse of the same token for downloading multiple documents, which could be a potential security vulnerability.

Styles have also been tweaked for MinePermitTable, Modal, and new styles for ExplosivesPermits have been created.

Re-ordered import statements to keep maintain consistency and readability across the code base.

matbusby-fw commented on services/core-web/src/components/Forms/ExplosivesPermit/MagazineFormNew.tsx:
```diff
@@ -0,0 +1,308 @@
+import React from "react";
```
Rather than add a bunch of feature flags, I opted to create "New" files that can be loaded conditionally.

matbusby-fw replied:
Apologies that it doesn't show a proper diff as a result.  Most of the changes are style related.

matbusby committed:
update snap

matbusby committed:
update snap

matbusby committed:
update snap

taraepp commented on services/core-web/src/components/mine/ExplosivesPermit/MineExplosivesPermitTable.tsx:
```diff
@@ -223,158 +229,186 @@ const MineExplosivesPermitTable: FC<RouteComponentProps & MineExplosivesPermitTa
       key: "addEditButton",
       align: "right",
-      render: (record) => {
+      render: (text, record) => {
```
I'd actually _not_ add a dataIndex here. If the data has no property "addEditButton", all it does is add a second redundant parameter to the render function.

matbusby-fw replied:
Ah, good point!

matbusby committed:
"Remove unnecessary defaultProps from DocumentTable component

The defaultProps object in DocumentTable.tsx was removed because it was not being used bring any value to the declared component. The methods: openModal and closeModal, defined in this object, were not applied in the component. It enhances readability by removing unused parts of code."

matbusby committed:
Remove unused dataIndex from ExplosivesPermitTable

taraepp commented on services/core-web/src/components/mine/ExplosivesPermit/ExplosivesPermit.tsx:
```diff
@@ -185,7 +200,8 @@ export const ExplosivesPermit: FC<ExplosivesPermitProps> = ({
         const initialValues = {};
-        props.documentContextTemplate.document_template.form_spec.forEach(
+        props.documentContextTemplate.document_template.form_spec.map(
+          // eslint-disable-next-line no-return-assign
```
This one

matbusby committed:
address PR comments
